server/Route.js

```diff
@@ -1,5 +1,55 @@
 const express = require('express');
+const db = require('./db/index');
 
 const router = express.Router();
 
+router.get('/:restaurant_id', (req, res) => {
+  const { restaurant_id } = req.params;
+
+  return db.ReservationsTable.findAll(
+    { where: { restaurant_id } },
+  )
+    .then(data => res.status(200).send(data))
+    .catch(err => res.status(500).send(err));
+});
+
+router.post('/:restaurant_id/:reservation_year/:reservation_month/:reservation_date/:reservation_hour/:reservation_min', (req, res) => {
+  const { restaurant_id, reservation_year, reservation_month, reservation_date, reservation_hour, reservation_min } = req.params;
+  const { party_size } = req.body;
+
+  return db.ReservationsTable.findOne(
+    {
+      where: {
+        restaurant_id,
+        reservation_year,
+        reservation_month,
+        reservation_date,
+        reservation_hour,
+        reservation_min,
+      },
+    },
+  )
+    .tap((reservation) => {
+      reservation.decrement('available_seats', { by: party_size });
```

Comment:
@LittleJennie 
@Sam-I-Am92 

This may be in the details of your implementation, but why do these three .taps need to happen separately (e.g., why can't I decrement available seats by party size and increment the total booking number) at the same time?

Reply:
I wasn't able to figure out a way to update these the increment and decrement within the same update function. I tried doing `db.ReservationTable.update({available_seats: db.ReservationTable.col('available_seats') + party_size, restaurant_total_booking_num: db.ReservationTable.col('restaurant_total_booking_num') + 1} {where: ...}), but it didn't work out correctly.